packages/parser/src/language/common/tokenBuilder.ts

```diff
@@ -11,6 +11,14 @@ export abstract class MermaidTokenBuilder extends DefaultTokenBuilder {
     this.keywords = new Set<string>(keywords);
   }
 
+  public override buildTokens(
+    grammar: GrammarAST.Grammar,
+    options?: TokenBuilderOptions | undefined
+  ): TokenVocabulary {
+    this.rearrangeRules(grammar.rules);
+    return super.buildTokens(grammar, options);
+  }
```

Comment:
I think we could use this approach; the `grammar` parameter would have the array of rules as it has been declared in the `.langium`, declared rules at the beginning and imported rules at the end. Then we could have a pivot rule (in this case it's `TitleAndAccessibilities`), and flip all other rules around it. After that, `super.buildTokens` should just sort them in terms of efficiency.

For example: `[1,2,3,4,5]` with pivot 3 should return `[4,5,3,1,2]`.

CC @weedySeaDragon

Reply:
Should we wait for this to be resolved?

This PR is good to go otherwise.

Reply:
I think we should wait. Since @weedySeaDragon has already created a PR showing a different approach #4910, it would be reasonable to wait for his response.